rest/zone_test.go

```diff
@@ -80,9 +90,36 @@ func TestPagination_disabled(t *testing.T) {
 	}
 }
 
-func mockAPI(t *testing.T) *httptest.Server {
+func TestZone_http_errors(t *testing.T) {
```

Comment:
I think one failure scenario is untested here: when `c.httpClient.Do(req)` in `Do` returns `err`.  This should return a nil `resp`.  Not a HTTP error, of course, so maybe it belongs in it's own test.